features/provider_management.feature

```diff
@@ -44,11 +42,46 @@ Feature: Case worker can manage providers
     And I click govuk checkbox 'LGFS'
     And I fill in 'Supplier number' with '1A234B'
     And I choose govuk radio 'Yes' for 'Is the provider VAT registered?'
-    Then the page should be accessible skipping 'aria-allowed-attr'
 
     When I click the button 'Save details'
     Then I should see 'Provider successfully created'
+
+    And I eject the VCR cassette
+
+  Scenario: A Provider manager creates a new firm with errors and corrects
+    When I insert the VCR cassette 'features/provider_management'
+    Given I am a signed in case worker provider manager
     Then the page should be accessible
+    When I click the link 'Providers'
+
+    When I click the link 'Add a provider'
+    Then I should be on the new provider page
+
+    When I choose govuk radio 'Firm' for 'Provider type'
+    And I click govuk checkbox 'LGFS'
+    And I fill in 'Supplier number' with '11111'
+
+    When I click the button 'Save details'
+    Then the following govuk error details should exist:
+      | field_type | field_locator | error_text | linked_id |
+      | field | Provider name| Enter a provider name | provider-name-field-error |
+      | field | Supplier number | Enter a valid LGFS supplier number | provider-lgfs-supplier-numbers-attributes-0-supplier-number-field-error |
+      | fieldset | Is the provider VAT registered? | Choose VAT registration state | provider-vat-registered-field-error |
+
+    # long version
```

Comment:
To be or not to be?

Reply:
yes, this is the question of preference i wanted to raise in the commits comment with you guys
> I prefer the more succinct cucumber table as a reusable
pattern for future unhappy path scenarios. However,
this could be more confusing during failures so would
like other devs views and preferences input.

What do you think?

Reply:
The tabular form is more understandable. What happens for a failure? Does it just indicate a failure without specifying which line? I don't think that is a problem, especially if there are just a few lines.

Reply:
its not too bad, it will fail on the first row and expectation but from my testing it is pretty easy to identify the failure